### PR TITLE
Attempt to somewhat finalize the format of messages

### PR DESCRIPTION
**Pull request by craumix**
Message get a new more modular format which also allows to improve internal passing of message content, and better blob handling when transfering messages.

### CODE REVIEW

File: pkg/types/room.go [hunk 1]
```diff
@@ -97,7 +97,10 @@ This only adds users, and can't remove users from peers.
 */
 func (r *Room) syncPeerLists() {
 	for _, peer := range r.Peers {
-		r.SendMessageToAllPeers(MessageTypeCmd, []byte("join "+peer.RIdentity.Fingerprint()), MessageContentInfo{})
+		r.SendMessageToAllPeers(MessageContent{
+			Type: MessageTypeCmd,
+			Data: []byte("join "+peer.RIdentity.Fingerprint()),
```
Comment (Meerschwein) [inline pkg/types/room.go:102]: Use the RoomCommand enum instead of join

File: pkg/types/message.go [hunk 2]
```diff
@@ -15,20 +15,24 @@ const (
...
-	Sender      string             `json:"sender"`
-	Time        time.Time          `json:"time"`
-	Type        MessageType        `json:"type"`
-	ContentInfo MessageContentInfo `json:"contentMeta,omitempty"`
+	Sender string    `json:"sender"`
+	Time   time.Time `json:"time"`
+}
+
+type MessageContent struct {
+	Type MessageType `json:"type"`
```
Comment (Meerschwein) [inline pkg/types/message.go:30]: MessageType should probably stay in MessageMeta

Comment (craumix) [reply pkg/types/message.go:30]: maybe rename to content type

File: pkg/types/room.go (hunk 1 above)
Comment (craumix) [reply pkg/types/room.go]: done

File: pkg/types/room.go [hunk 3]
```diff
@@ -99,7 +100,7 @@ func (r *Room) syncPeerLists() {
 	for _, peer := range r.Peers {
 		r.SendMessageToAllPeers(MessageContent{
 			Type: MessageTypeCmd,
-			Data: []byte("join "+peer.RIdentity.Fingerprint()),
+			Data: []byte(string(RoomCommandJoin) + peer.RIdentity.Fingerprint()),
```
Comment (Meerschwein) [inline pkg/types/room.go]: I think we need a space here

Comment (craumix) [reply pkg/types/room.go]: gofmt says its fine ;)

Comment (Meerschwein) [reply pkg/types/room.go]: `Data: []byte(string(RoomCommandJoin) + " " + peer.RIdentity.Fingerprint()),`
We need the space as delimiter
Or add a space to the RoomCommands

**Pull request merged**